src/Microsoft.OData.Core/UriParser/Binders/RangeVariableBinder.cs

```diff
@@ -24,7 +24,11 @@ internal static SingleValueNode BindRangeVariableToken(RangeVariableToken rangeV
         {
             ExceptionUtils.CheckArgumentNotNull(rangeVariableToken, "rangeVariableToken");
 
-            RangeVariable rangeVariable = state.RangeVariables.SingleOrDefault(p => p.Name == rangeVariableToken.Name);
+            // We are using FirstOrDefault instead of SingleOrDefault since we can have >1 $it rangeVariable in the Stack.
+            // If we added another $it rangeVariable to the Stack (when binding the expandToken or selectToken),
+            // We create the RangeVariableReferenceNode using the added $it rangeVariable.
+            // ASSUMPTION: Max 2 $it rangeVariables can be added to the Stack.

```

Comment:
The method is called for all rangeVariables. 
Initially, a `QueryToken` could only have one `rangeVariable`. Since we have added another `$it` rangeVariable in `SelectTermToken` and `ExpandTermToken`, we can have a max of 2 $it rangeVariables in the stack.

Reply:
I refractored the code to have only 1 $it range variable.
#Resolved